### PR TITLE
Add/remove Async suffix from method names (fixes S4261)

### LINKED ISSUE #952: Address S4261: Methods should be named according to their synchronicities
Address existing violations of [S4261: Methods should be named according to their synchronicities](https://rules.sonarsource.com/csharp/RSPEC-4261) in the codebase and set severity to `Warning` in `Steeltoe.Debug.ruleset` and `Steeltoe.Release.ruleset`.

To find existing violations, enable the rule (see above) and rebuild `src/Steeltoe.All.sln` to make them appear in the Output window.

To address the violations, choose from the following on a case-by-case basis:
- Add or remove the "Async" suffix
- ~Suppress the violation in code using `#pragma warning disable/restore`, preceded by a justification comment if not obvious~

### PR DESCRIPTION
**Pull request by bart-vmware**
Fixes #952.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [ ] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs [hunk 1]
```diff
@@ -925,13 +925,13 @@ protected internal string FetchAccessToken()
...
 
-        return HttpClientHelper.GetAccessToken(settings.AccessTokenUri, settings.ClientId, settings.ClientSecret, settings.Timeout,
+        return HttpClientHelper.GetAccessTokenAsync(settings.AccessTokenUri, settings.ClientId, settings.ClientSecret, settings.Timeout,
             settings.ValidateCertificates, httpClient, logger).GetAwaiter().GetResult();
     }
 
     // fire and forget
 #pragma warning disable S3168 // "async" methods should not return "void"
-    protected internal async void RefreshVaultTokenAsync(object state)
+    protected internal async void RefreshVaultToken(object state)
```
Comment (TimHess) [inline src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs:934]: While it doesn't return `Task`, shouldn't it retain `async` anyway given that it is async?

Comment (TimHess) [reply src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs:934]: * This is not the only instance, but I'll refrain from commenting on any others until we resolve here

File: src/CircuitBreaker/test/HystrixBase.Test/HystrixCommandTest.cs [hunk 2]
```diff
@@ -849,7 +849,7 @@ public async Task TestRejectedThreadWithNoFallback()
             f = command1.ExecuteAsync(); // Running
             Time.Wait(50); // Let first start
             f2 = command2.ExecuteAsync(); // In Queue
-            await command3.ExecuteAsync(); // Start, queue rejected
+            await command3.ExecuteAsync(); // StartAsync, queue rejected
```
[the remark this answers is not in the document]
Comment (bart-vmware) [reply src/CircuitBreaker/test/HystrixBase.Test/HystrixCommandTest.cs:852]: Nice catch! I must have missed that one while reviewing proposed changes in comments as part of rename. I'll revert this one.

File: src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs [hunk 3]
```diff
@@ -93,7 +93,7 @@ public override void Send(string exchange, string routingKey, IMessage message,
                 {
                     _cancellationTokenSource = new CancellationTokenSource();
                     TimeSpan delay = next.Value - DateTime.Now;
-                    _scheduledTask = Task.Run(() => ReleaseBatches(delay, _cancellationTokenSource.Token), _cancellationTokenSource.Token);
+                    _scheduledTask = Task.Run(() => ReleaseBatchesAsync(delay, _cancellationTokenSource.Token), _cancellationTokenSource.Token);
```
Comment (TimHess) [inline src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs:96]: Should there also be an `await` on this line? cc @dtillman @hananiel

File: src/Common/test/Common.Http.Test/HttpClientHelperTest.cs [hunk 4]
```diff
@@ -162,9 +162,9 @@ public void GetRequestMessage_CreatesCorrectMessage_WithBasicAuth()
     [Fact]
     public void GetAccessToken_ThrowsNulls()
```
[the remark this answers is not in the document]
Comment (bart-vmware) [reply src/Common/test/Common.Http.Test/HttpClientHelperTest.cs:163]: That's hard to say for sure. Assuming the exception is thrown before the first await point, there's no need to actually await the call. The same concept applies to `yield returns`. So if the intent of this test is to throw early, then it is correct. But the test is named quite poorly, so we can't know.

```c#
var task = HttpClientHelper.GetAccessToken(...);  // throw here? (seems most correct to me)
await task; // or here?
```

File: src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs (hunk 1 above)
Comment (bart-vmware) [reply src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs:934]: I've removed the Async suffix in all three suggested cases, for the next reason. The purpose of the `async` keyword in the method declaration is only for the method body itself. It is unobservable by callers, they see just `void` or `Task`. So from a callers perspective, this method has nothing to do with asynchronous execution, which is why adding the prefix would be wrong.

File: src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs (hunk 3 above)
Comment (bart-vmware) [reply src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs:96]: Yes, I suspect it should. The `ReleaseBatchesAsync` method contains await points. So any exception thrown after the first await point goes unnoticed. Maybe even crash the process, although I'm not entirely sure about that as it depends on the execution context.

Comment (TimHess) [reply src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs:96]: This concern will be revisited in another issue that I'll create momentarily

**Review by TimHess — APPROVED** [review]
These changes are an improvement that have highlighted some problems that will be fixed in another PR

File: src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs (hunk 3 above)
Comment (bart-vmware) [reply src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs:96]: Resolving, this is trakced at #983.

**Pull request merged**